systemtests/src/test/java/io/enmasse/systemtest/common/upgrade/UpgradeTest.java

```diff
@@ -23,45 +33,172 @@
     // Help methods
     ///////////////////////////////////////////////////////////////////////////////////////////////////////
 
-    private void applyEnmasseVersion(Path templatePaths, boolean downgrade) throws Exception {
+    private void doTestUpgrade(boolean isAnsible) throws Exception {
+        if (isAnsible) {
+            installEnmasseAnsible(Paths.get(Environment.getInstance().getStartTemplates()), false);
+        } else {
+            installEnmasseBundle(Paths.get(Environment.getInstance().getStartTemplates()), startVersion);
+        }
+
+        createAddressSpaceCMD(kubernetes.getInfraNamespace(), "brokered", "brokered", "brokered-single-broker", null, "v1alpha1");
+        createAddressSpaceCMD(kubernetes.getInfraNamespace(), "standard", "standard", "standard-unlimited-with-mqtt", null, "v1alpha1");
+        Thread.sleep(30_000);
+
+        createUserCMD(kubernetes.getInfraNamespace(), "test-brokered", "test", "brokered", "v1alpha1");
+        createUserCMD(kubernetes.getInfraNamespace(), "test-standard", "test", "standard", "v1alpha1");
+        Thread.sleep(30_000);
+
+        createAddressCMD(kubernetes.getInfraNamespace(), "brokered-queue", "brokered-queue", "brokered", "queue", "brokered-queue", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "brokered-topic", "brokered-topic", "brokered", "topic", "brokered-topic", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-queue", "standard-queue", "standard", "queue", "standard-large-queue", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-topic", "standard-topic", "standard", "topic", "standard-small-topic", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-anycast", "standard-anycast", "standard", "anycast", "standard-small-anycast", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-multicast", "standard-multicast", "standard", "multicast", "standard-small-multicast", "v1alpha1");
+        Thread.sleep(30_000);
+
+        waitUntilDeployed(kubernetes.getInfraNamespace());
+        Thread.sleep(60_000);
+
+        assertTrue(sendMessage("brokered", new RheaClientSender(), new UserCredentials("test-brokered", "test"), "brokered-queue", "pepa", MESSAGE_COUNT, true));
+        assertTrue(sendMessage("standard", new RheaClientSender(), new UserCredentials("test-standard", "test"), "standard-queue", "pepa", MESSAGE_COUNT, true));
+
+        if (isAnsible) {
+            installEnmasseAnsible(Paths.get(Environment.getInstance().getUpgradeTemplates()), true);
+        } else {
+            upgradeEnmasseBundle(Paths.get(Environment.getInstance().getUpgradeTemplates()));
+        }
+        Thread.sleep(30_000);
+
+        if (!startVersion.equals("install")) {
+            // KW: I am seeing an issue in standard where after a Broker with queue depth comes up, sometimes messages
```

Comment:
I have a working theory about what we might be seeing.  I suspect a broker issue but don't have sufficient to raise a Apache Artemis JIRA at the moment.